src/dash-table/syntax-tree/lexeme/relational.ts

```diff
@@ -44,49 +44,80 @@ const LEXEME_BASE = {
     type: LexemeType.RelationalOperator
 };
 
+const containsEval = (lhs: any, rhs: any, relOp: string): boolean =>
+    relOp[0] == 'i'
+        ? lhs.toString().toUpperCase().indexOf(rhs.toString().toUpperCase()) !==
+          -1
+        : lhs.toString().indexOf(rhs.toString()) !== -1;
+
+const equalEval = (lhs: any, rhs: any, relOp: string): boolean =>
+    isNumeric(lhs) && isNumeric(rhs)
+        ? +lhs === +rhs
+        : relOp[0] == 'i'
+        ? lhs.toString().toUpperCase() === rhs.toString().toUpperCase()
+        : lhs === rhs;
+
+const fnEval = (
+    fn: (lhs: any, rhs: any) => boolean,
+    lhs: any,
+    rhs: any,
+    relOp: string
+): boolean =>
+    relOp[0] == 'i'
+        ? fn(lhs.toString().toUpperCase(), rhs.toString().toUpperCase())
+        : fn(lhs, rhs);
+
 export const contains: IUnboundedLexeme = R.merge(
     {
         evaluate: relationalEvaluator(
-            ([op, exp]) =>
-                !R.isNil(exp) &&
-                !R.isNil(op) &&
-                (R.type(exp) === 'String' || R.type(op) === 'String') &&
-                op.toString().indexOf(exp.toString()) !== -1
+            ([lhs, rhs, relOp]) =>
+                !R.isNil(rhs) &&
+                !R.isNil(lhs) &&
+                (R.type(rhs) === 'String' || R.type(lhs) === 'String') &&
+                containsEval(lhs, rhs, relOp)
         ),
         subType: RelationalOperator.Contains,
-        regexp: /^((contains)(?=\s|$))/i,
+        regexp: /^((i|s)?contains)(?=\s|$)/i,
+        regexpFlags: 2,
```

Comment:
regex match group for usage in https://github.com/plotly/dash-table/pull/893/files#r622660202